### PR TITLE
Slaughter Demon powers can now be used by any mob

### PR DESCRIPTION
**Pull request by ghost**
Cleaned up the blood jaunt code to use only two vars (bloodcrawl and holder) and gave said vars to /mob/living.

bloodcrawl = 1 lets you move in/out of blood pools, bloodcrawl = 2 lets you devour mobs

Eating the demon's heart now gives you bloodcrawl = 1

Removed colons from the slaughter demon spawner, sorry again for those.

### CODE REVIEW

**Comment by Supermichael777** [comment]
vampires when? also SHENANIGANS.

**Comment by Incoming5643** [comment]
Wizards might want to spawn a demon just to try and kill it themselves

**Comment by optimumtact** [comment]
May as well go all the way up to atmo/movable

:laughing:

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 1]
```diff
@@ -27,12 +27,8 @@
 	melee_damage_lower = 30
 	melee_damage_upper = 30
 	see_in_dark = 8
+	bloodcrawl = 2
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: can you make these defines?
```
#define BLOODCRAWL          1
#define BLOODCRAWL_EAT 2
```

File: code/modules/mob/living/bloodcrawl.dm [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+/mob/living/proc/phaseout(obj/effect/decal/cleanable/B)
+	var/mob/living/kidnapped = null
+	var/turf/mobloc = get_turf(src.loc)
+	var/turf/bloodloc = get_turf(B.loc)
+	if(Adjacent(bloodloc))
+		src.notransform = TRUE
+		spawn(0)
+			src.visible_message("[src] sinks into the pool of blood.")
+			playsound(get_turf(src), 'sound/magic/enter_blood.ogg', 100, 1, -1)
+			var/obj/effect/dummy/slaughter/holder = new /obj/effect/dummy/slaughter( mobloc )
```
Comment (RemieRichards) [inline code/modules/mob/living/bloodcrawl.dm]: can you look into pooling these slaughter holders?
They don't have very many variables so they should be an easy one.

Comment (KorPhaeron) [inline code/modules/mob/living/bloodcrawl.dm]: I've never pooled objects before, off the top of your head is there a good example I can look at in the code?

Comment (RemieRichards) [inline code/modules/mob/living/bloodcrawl.dm]: Well, swap new for PoolOrNew(), look up the doc on that proc
and swap qdel for ReturnToPool() (OR, and this is probably better, override Destroy() to return QDEL_HINT_PLACE_IN_POOL)

a good example would probably be fire hotspots, or virtual speakers.

**Comment by daman997** [comment]
slaughter people when

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 3]
```diff
@@ -172,4 +52,10 @@ obj/effect/dummy/slaughter/relaymove(mob/user, direction)
...
 	icon = 'icons/obj/surgery.dmi'
 	icon_state = "heart-on"
-	origin_tech = "combat=5;biotech=8"
\ No newline at end of file
+	origin_tech = "combat=5;biotech=8"
+
+/obj/item/weapon/demonheart/attack_self(mob/living/user)
+	visible_message("[user] feasts upon the [src].")
+	user << "You absorb some of the demon's power!"
+	user.bloodcrawl = 1
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: If you've already eaten a heart, you should get bloodcrawl = 2, and if you're at bloodcrawl =2 already you shouldn't be able to eat it.

**Comment by daman997** [comment]
anonus that is fucking badass

**Comment by Rinoka** [comment]
I agree with slaughter demon hearts being an incrementer, not a just +1. Just make something hilarious that deincrements it, like syringes of demon blood and holy water.

**Comment by KorPhaeron** [comment]
@RemieRichards 

Everything should be ready now

**Pull request merged**